Mathlib/Algebra/Group/Subgroup/Defs.lean

```diff
@@ -73,6 +73,18 @@ class NegMemClass (S : Type*) (G : outParam Type*) [Neg G] [SetLike S G] : Prop
 
 export NegMemClass (neg_mem)
 
+/-- Typeclass for substructures `s` such that `s ∪ -s = G`. -/
+class HasMemOrNegMem {S G : Type*} [AddCommGroup G] [SetLike S G] (s : S) : Prop where
+  mem_or_neg_mem (s) (a : G) : a ∈ s ∨ -a ∈ s
+
+/-- Typeclass for substructures `s` such that `s ∪ s⁻¹ = G`. -/
+@[to_additive]
+class HasMemOrInvMem {S G : Type*} [CommGroup G] [SetLike S G] (s : S) : Prop where
```

Comment:
Perhaps you'd like to generalise this further too, but having comm here feels particularly strange to me